src/types/metaFactory-types/lens.ts

```diff
@@ -0,0 +1,102 @@
+import _ from 'lodash';
+
+import { artistId } from '../../utils/identifiers';
+import { formatAddress } from '../address';
+import { getFactoryId } from '../chain';
+import { ArtistIdExtractorTypes, ArtistNameExtractorTypes, IdExtractorTypes, TitleExtractorTypes } from '../fieldExtractor';
+import { MetaFactoryType } from '../metaFactory';
+import { NFTContractTypeName } from '../nft';
+
+const type: MetaFactoryType = {
+  newContractCreatedEvent: 'CollectNFTDeployed',
+  metadataAPI: async (events, clients, metaFactory) => {
+    if (events.length === 0){
+      return {}
+    }
+
+    const maxEventBlockNumber = events.reduce((maxBlock, event) => Math.max(maxBlock, event.blockNumber), 0)
+
+    const profileCreatedEvents = await clients.evmChain[metaFactory.chainId]
+      .getEventsFrom(
+        metaFactory.startingBlock!,
+        maxEventBlockNumber.toString(),
+        events.map(event => {
+          const profileId = ethers.BigNumber.from(event.args!.profileId).toNumber();
+          return {
+            address: metaFactory.address,
+            filter: `ProfileCreated`,
+            filterArgs: [profileId]
+          }
+        }))
+
+    const profileCreatedEventByProfileId = _.keyBy(
+      profileCreatedEvents,
+      event => event.args!.profileId!.toNumber()
+    )
+
+    const profilesByCollectNft = _.keyBy(
+      events.map(event => {
+        const profileId = ethers.BigNumber.from(event.args!.profileId).toNumber()
+        const profileCreatedEvent = profileCreatedEventByProfileId[profileId] as any;
+        
+        return {
+          collectNFT: event.args!.collectNFT,
+          profile: {
+            handle: profileCreatedEvent.args.handle,
+            ownedBy: profileCreatedEvent.args.to,
+            id: profileId,
+          }
+        }
+      }),
+      'collectNFT',
+    )
+
+    return profilesByCollectNft
+
+  },
+  creationMetadataToNftFactory(
+    event,
+    autoApprove,
+    metaFactory,
+    factoryMetadata: { 
+      [collectNFT: string]: { 
+        profile: { 
+          handle: string;
+          ownedBy: string;
+          id: string;
+        } 
+      } 
+    })
+  {
+    const apiMetadata = factoryMetadata[event.args!.collectNFT];
+
+    return {
+      id: getFactoryId(metaFactory.chainId, event.args.collectNFT),
+      address: formatAddress(event.args.collectNFT),
+      approved: autoApprove,
+      autoApprove: autoApprove,
+      chainId: metaFactory.chainId,
+      contractType: NFTContractTypeName.default,
+      platformId: metaFactory.platformId,
+      standard: metaFactory.standard,
+      startingBlock: `${parseInt(event.blockNumber) - 1}`,
+      typeMetadata: {
+        overrides: {
+          extractor: {
+            id: IdExtractorTypes.USE_TITLE_EXTRACTOR,
+            title: TitleExtractorTypes.METADATA_NAME,
+            artistName: ArtistNameExtractorTypes.USE_ARTIST_NAME_OVERRIDE,
+            artistId: ArtistIdExtractorTypes.USE_ARTIST_ID_OVERRIDE,
```

Comment:
do we still need these extractors given that we're overriding them below?

Reply:
Yes because the multi-track-multi-print mappers use these resolvers, so we have to use these values to specify that the resolvers should use the override value